src/with-unique.js

```diff
@@ -1,14 +1,56 @@
-import { HTMLElement, sym, uniqueId } from './util';
+import { customElements, dashCase, HTMLElement, keys, sym } from './util';
+import { define } from './define';
 
 const _is = sym();
+const baseFunctionStatics = keys(function () {});
 
-export function withUnique (Base = HTMLElement) {
-  return class extends Base {
-    static get is () {
-      return this[_is] || (this[_is] = `x-${uniqueId().substring(0, 8)}`);
+let suffix = 0;
+
+function extendStatics (Ctor, Base) {
+  keys(Base).forEach(name => {
+    if (baseFunctionStatics.indexOf(name) > -1) {
+      return;
+    }
+    const descriptor = Object.getOwnPropertyDescriptor(Base, name);
+    if (descriptor) {
+      Object.defineProperty(Ctor, name, descriptor);
     }
-    static set is (is) {
-      this[_is] = is;
+  });
+}
+
+function formatName (prefix, suffix) {
+  return (
+    prefix.indexOf('-') === -1 ? `x-${prefix}` : prefix
+  ) + (
+    suffix ? `-${suffix}` : ''
+  );
+}
+
+function generateName (Ctor, hint) {
+  const prefix = hint || dashCase(Ctor.name) || 'element';
+  while (customElements.get(formatName(prefix, suffix))) {
+    suffix++;
+  }
+  return formatName(prefix, suffix++);
+}
+
+export function withUnique (Base = HTMLElement) {
+  function Ctor () {
+    const { constructor } = this;
+    define(constructor);
```

Comment:
I think a good compromise here might be to bake this into `define()`. It's still explicit, but gives you simpler ergonomics from being able to use the class name if `static is` is not defined. Mixins are opt-in anyways.

In fact, if we made everything decorators / functions, then this could be baked in without the hacks:

```js
import { component, define } from 'skatejs';

@component
@define
class HelloWorld extends HTMLElement {}
```